### PR TITLE
[UPD] base_wamas_ubl: update address export

### PR DESCRIPTION
**Pull request by thienvh332**

### CODE REVIEW

File: base_wamas_ubl/lib/wamas/utils.py [hunk 1]
```diff
@@ -217,6 +217,55 @@ def convert_unit_code(key, val):
...
+        ),
+        "AdditionalStreetName": dict_item.get(
+            "DespatchAdvice.cac:DeliveryCustomerParty."
+            "cac:Party.cac:PostalAddress.cbc:AdditionalStreetName"
+        ),
+    }
+
+
+def get_Adrs_Name(*args):
+    a = args[0]
```
Comment (nilshamerlinck) [inline base_wamas_ubl/lib/wamas/utils.py]:
```suggestion
def get_Adrs_Name(a):
```

Comment (nilshamerlinck) [reply base_wamas_ubl/lib/wamas/utils.py]: same for others, let's keep it simple

File: base_wamas_ubl/lib/wamas/test_utils.py [hunk 2]
```diff
@@ -29,6 +41,62 @@ def testDict2wamas(self):
...
+        result = get_Adrs_Name2(*args)
+        self.assertEqual(result, "YourCompany")
+        result = get_Adrs_Name3(*args)
+        self.assertEqual(result, "Department_1")
+        result = get_Adrs_Name4(*args)
+        self.assertEqual(result, "250 Executive Park Blvd, Suite 3400")
+        result = get_Adrs_Adr(*args)
+        self.assertEqual(result, "AdditionalStreetName_1")
+
+    def testGetAddressNameReception(self):
```
Comment (nilshamerlinck) [inline base_wamas_ubl/lib/wamas/test_utils.py]: you can remove this test + xml file, it doesn't add any value

File: base_wamas_ubl/lib/wamas/test_utils.py [hunk 3]
```diff
@@ -29,6 +41,62 @@ def testDict2wamas(self):
             output = dict2wamas(ast.literal_eval(str_dict_input), telegram)
             self.assertEqual(output, expected_output)
 
+    def testGetAddressNamePicking(self):
```
Comment (nilshamerlinck) [inline base_wamas_ubl/lib/wamas/test_utils.py:44]: you can keep just one of the xml file, to test get_address_elements

File: base_wamas_ubl/lib/wamas/test_utils.py [hunk 4]
```diff
@@ -29,6 +41,62 @@ def testDict2wamas(self):
...
+            address_elements,
+            {
+                "ContactName": "Chester Reed",
+                "PartyName": "YourCompany",
+                "Department": "Department_1",
+                "StreetName": "250 Executive Park Blvd, Suite 3400",
+                "AdditionalStreetName": "AdditionalStreetName_1",
+            },
+        )
+        args = (address_elements,)
```
Comment (nilshamerlinck) [inline base_wamas_ubl/lib/wamas/test_utils.py]:
```suggestion
```

File: base_wamas_ubl/lib/wamas/test_utils.py [hunk 5]
```diff
@@ -29,6 +41,62 @@ def testDict2wamas(self):
...
+        result = get_Adrs_Name(*args)
```
Comment (nilshamerlinck) [inline base_wamas_ubl/lib/wamas/test_utils.py:61]:
```suggestion
        result = get_Adrs_Name(address_elements)
```

Comment (nilshamerlinck) [reply base_wamas_ubl/lib/wamas/test_utils.py:61]: let's keep things simple

File: base_wamas_ubl/lib/wamas/test_utils.py [hunk 6]
```diff
@@ -29,6 +41,62 @@ def testDict2wamas(self):
...
+        self.assertEqual(result, "Chester Reed")
```
Comment (nilshamerlinck) [inline base_wamas_ubl/lib/wamas/test_utils.py]:
```suggestion
        self.assertEqual(get_Adrs_Name(address_elements), "Chester Reed")
```

Comment (nilshamerlinck) [reply base_wamas_ubl/lib/wamas/test_utils.py]: let's keep things simple
I usually use tests to understand what are the specs for a function

Comment (nilshamerlinck) [reply base_wamas_ubl/lib/wamas/test_utils.py]: same bellow

File: base_wamas_ubl/tests/samples/UBL2WAMAS-SAMPLE_KRETK_KRETP.wamas [hunk 7]
```diff
@@ -1,6 +1,6 @@
-ODOO      WAMAS     00000120230501020000KRETK0051000130377              HOST                     STDMAN0001040         MyCompany                                                                                                                                                                      1839 Arbor Way                          California                              95380     Turlock                                                                         US  (623)-853-7197                                                        wood.corner26@example.com                                                                                                                                                          20230427000000
```
Comment (nilshamerlinck) [inline base_wamas_ubl/tests/samples/UBL2WAMAS-SAMPLE_KRETK_KRETP.wamas:1]: AdditionalStreetName_1 doesn't appear here?

File: base_wamas_ubl/lib/wamas/test_utils.py [hunk 8]
```diff
@@ -29,6 +41,30 @@ def testDict2wamas(self):
...
+            },
+        )
+        self.assertEqual(get_Adrs_Name(address_elements), "Chester Reed")
+        self.assertEqual(get_Adrs_Name2(address_elements), "YourCompany")
+        self.assertEqual(get_Adrs_Name3(address_elements), "Department_1")
+        self.assertEqual(
+            get_Adrs_Name4(address_elements), "250 Executive Park Blvd, Suite 3400"
+        )
+        self.assertEqual(get_Adrs_Adr(address_elements), "AdditionalStreetName_1")
+
```
Comment (nilshamerlinck) [inline base_wamas_ubl/lib/wamas/test_utils.py:67]: can you add one more test case not based on a xml file, but based on a given {}
that matches the example given in the specs:  Duffard Pascal
https://docs.google.com/document/d/1JzXZOgPSkvVYP9naE9qA_wIY2T9ElbArvN71f_4cbik/edit

the goal is to have a different example to see different expectations

File: base_wamas_ubl/tests/samples/UBL2WAMAS-SAMPLE_KRETK_KRETP.wamas (hunk 7 above)
Comment (thienvh332) [reply base_wamas_ubl/tests/samples/UBL2WAMAS-SAMPLE_KRETK_KRETP.wamas:1]: > AdditionalStreetName_1 doesn't appear here?

I haven't added it to the base_wamas_ubl/tests/samples/UBL2WAMAS-SAMPLE_KRETK_KRETP-DESPATCH_ADVICE.xml file yet so it's not there yet. I will add it.

File: base_wamas_ubl/lib/wamas/test_utils.py [hunk 9]
```diff
@@ -29,6 +41,56 @@ def testDict2wamas(self):
...
+    def testGetAddressNameFromDict(self):
+        dict_data = {
```
Comment (nilshamerlinck) [inline base_wamas_ubl/lib/wamas/test_utils.py]: we don't need this, just feed `address_elements` directly (we don't need to test `get_address_elements` twice)

**Review by nilshamerlinck — APPROVED** [review]

**Pull request merged**